x/gov/codec.proto

```diff
@@ -0,0 +1,93 @@
+syntax = "proto3";
+
+package gov;
+
+import "github.com/gogo/protobuf/gogoproto/gogo.proto";
+import "github.com/iov-one/weave/coin/codec.proto";
+import "google/protobuf/timestamp.proto";
+
+// Electorate defines who may vote in an election. This same group can be used in many elections
+// and is stored for re-use
+message Electorate {
+    // Participants defines a list of all signatures that are allowed to participate in a vote
+    repeated Participant participants = 1 [(gogoproto.nullable) = false];
+}
+
+// Participant clubs together a signature with a weight. The greater the weight
+// the greater the power of a participant.
+message Participant {
+    bytes signature = 1 [(gogoproto.casttype) = "github.com/iov-one/weave.Address"];
+    uint32 power = 2 [(gogoproto.casttype) = "Weight"];
+}
+
+// Election Rules defines how an election is run. A proposal must be voted upon via a pre-defined ruleset.
+message ElectionRules {
+    // Human readable title.
+    string title = 1;
+    // Reference to the electorate to define the group of possible voters.
+    bytes electorate_id = 2;
+    // Duration how long the voting period will take place.
+    uint32 voting_period_hours = 3;
+    // Required Vote this is percentage of all eligible voters, not the ones who voted.
+    // (integer in steps of 0.01%, so 10000 is 100%. should disallow anything below 5001)
+    uint32 required_votes = 4;
```

Comment:
if 0.1% steps are sufficient, we can make this a per mille value (‰), which makes thinking and writing about it a bit more natural.

Reply:
I proposed .01% for 50.01% and 66.67%
But if others agree, I am okay with 50.1% and 66.7% as cut-offs as well.
Full percentage point seemed a bit steep to me

Reply:
Interesting. So 200 vs. 100 votes will never reach the 2/3 requirement, no matter if it is expressed as 66.7 %, 66.67% or 66.6666667%.

In this case, I'd move from floating point math to fractions and express `required_votes` as `uint32 numerator` and `uint32 denominator`. Then you require

```
yes / voted >= numerator / denominator
<=>
(yes*denominator) / (voted*denominator) >= (numerator*voted) / (denominator*voted)
<=>
(yes*denominator) >= (numerator*voted)
```

wich works nicely with integer math, e.g. using the above example again

```
(200*3) >= (2*300)
```

Reply:
@alpe I like this proposal by Simon above. I would just change it from >= to >. If the vote is 50-50 should it pass or fail? (If answer is pass, then >= is good).

So, `numerator: 1, denominator: 2` => `> 50%`
So, `numerator: 2, denominator: 3` => `> 66.666..%`
So, `numerator: 6273, denominator: 10000` => `> 62.73%`

This is more flexible and easy to understand and do math with.

Reply:
I think it should be `>=`. A 50%+ requirement can be expressed as `numerator: 500000001, denominator: 1000000000`

Reply:
Hmm, to be honest I am no to sure about the `>` vs. `>=` question. I don't find good resources.

Reply:
Two decimals after the comma as originally proposed are fine for me. I don't know if we need this higher precision but I am more used to set the comma as with currencies.

Reply:
The fractional math is indeed more important for the `>=` calculation since = 2/3 can easiliy be reached in fractional math when they are not reached in numeric math. For the `>` case it is much harder to come up with such a case.

Reply:
oh, I like the new proposal. I will update the protobuf. very good work!
In terms of `>=` vs `>`, I think it the second is easier to communicate and to setup. In natural language we would use more than half of the votes....